### PR TITLE
K1J-591: Implemented general print integration to certificate print service

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

**Comment by mhornfeldt** [comment]
Have not tested with CPS will add what is needed for serializing when able to test

File: certificate-service/integration-certificate-print-service/build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+group = 'se.inera.intyg.certificateservice'
+version = '0.0.1-SNAPSHOT'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.apache.pdfbox:pdfbox:3.0.2'
```
Comment (mww-sopra-steria) [inline certificate-service/integration-certificate-print-service/build.gradle:13]: Do we need to use pdf-box in this module? If we do then we shouldn't hardcode version.

File: certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    try {
+      return restClient
+          .post()
+          .uri(printCertificateServiceUrl + "/api/print/" + certificateId)
+          .header(LOG_TRACE_ID_HEADER, MDC.get(TRACE_ID_KEY))
+          .header(LOG_SESSION_ID_HEADER, MDC.get(SESSION_ID_KEY))
+          .body(request)
+          .retrieve()
+          .body(PrintCertificateResponseDTO.class);
+    } catch (HttpClientErrorException ex) {
```
Comment (mww-sopra-steria) [inline certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java:42]: Do we need to catch? Why return null if status is NOT_FOUND?

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Looks really good! Nice work!

A few questions/comments to look at before merging.

File: certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java (hunk 2 above)
Comment (mhornfeldt) [reply certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java:42]: used same pattern as intygsadmin but can update

Comment (mww-sopra-steria) [reply certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java]: I don't think we should return null, suggest that we don't catch any exceptions.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]

**Review by mww-sopra-steria — APPROVED** [review]

**Pull request merged**